### PR TITLE
emitHalideExpr: print parentheses around compound nested expressions

### PR DESCRIPTION
**Pull request by skimo-openhub**
Despite what the original comment claims, the expression in the output
that corresponds to a variable in the input may not be an identifier
by itself.  Since the variable may appear in a Halide expression
at any position (e.g., as an argument to "-" or "*"), parentheses
need to be put around the expression if it is anything other than
an identifier or a non-negative number.

### CODE REVIEW

**Comment by skimo-openhub** [comment]
Still waiting for confirmation that it actually fixed #179.

**Comment by skimo-openhub** [comment]
I removed the annotation that it fixes #179.  It should be an improvement, even if it doesn't.

File: src/core/polyhedral/cuda/codegen.cc [hunk 1]
```diff
@@ -495,22 +495,33 @@ isl::multi_aff makeMultiAffAccess(
...
       auto pwAff = tc::polyhedral::detail::makeAffFromMappedExpr(
           Halide::Expr(op), context);
       // Use a temporary isl::ast_build to print the expression.
       // Ideally, this should use the build at the point
       // where the user statement was created.
       auto astBuild = isl::ast_build::from_context(pwAff.domain());
-      auto s = astBuild.expr_from(pwAff).to_C_str();
+      auto expr = astBuild.expr_from(pwAff);
+      auto s = expr.to_C_str();
+      if (!is_identifier_or_nonnegative_integer(expr))
```
Comment (nicolasvasilache) [inline src/core/polyhedral/cuda/codegen.cc]: braces plz

**Comment by nicolasvasilache** [comment]
@caffe2bot retest this please

**Pull request merged**